src/make.js

```diff
@@ -0,0 +1,11 @@
+module.exports = function make(...args) {
```

Comment:
@warryr we need to discuss this function. It seems not all cases are considered.